### PR TITLE
refactor(presenter): utilize vike's client-side navigation

### LINKED ISSUE #105: 🔧 [Refactor] utilize vike.naviagte()
## 🔧 Refactor

utilize vike.naviagte() to use client navigation instead of server side navigation.
See: https://vike.dev/navigate

### PR DESCRIPTION
**Pull request by mahula**
## 🍰 Pullrequest
To benefit from Vike's possibilities for a lightweight and smooth user flow, we utilize its client-side navigation

### Issues
- fixes #105 

### Todo

- [X] None

### CODE REVIEW

File: presenter/src/components/inputs/MainButton.vue [hunk 1]
```diff
@@ -44,7 +46,7 @@ const props = withDefaults(
      */
     isLoading?: boolean
   }>(),
-  { variant: 'primary', size: 'medium', isLoading: false },
+  { href: undefined, variant: 'primary', size: 'medium', isLoading: false },
```
Comment (ulfgebhardt) [inline presenter/src/components/inputs/MainButton.vue:49]: is this needed or implicit?

File: presenter/src/components/inputs/MainButton.vue [hunk 2]
```diff
@@ -84,8 +86,15 @@ const setColor = computed(() => {
   }
 })
 
+const hasHref = computed(() => {
+  return props.href !== undefined
+})
+
 const onClick = () => {
   emit('click', 1)
```
Comment (ulfgebhardt) [inline presenter/src/components/inputs/MainButton.vue:94]: emit the href value? or what the 1 is for?

File: presenter/src/components/inputs/MainButton.vue [hunk 3]
```diff
@@ -84,8 +86,15 @@ const setColor = computed(() => {
...
+  if (hasHref.value) {
```
Comment (ulfgebhardt) [inline presenter/src/components/inputs/MainButton.vue:95]: if you move the check here you will not need the `as string`

File: presenter/src/components/inputs/MainButton.vue [hunk 4]
```diff
@@ -84,8 +86,15 @@ const setColor = computed(() => {
...
+    navigate(props.href as string)
+  }
```
Comment (ulfgebhardt) [inline presenter/src/components/inputs/MainButton.vue:97]:
```suggestion
  if (props.href) {
    navigate(props.href)
  }
```

File: presenter/src/components/inputs/MainButton.vue (hunk 1 above)
Comment (mahula) [reply presenter/src/components/inputs/MainButton.vue:49]: It is required.

File: presenter/src/components/inputs/MainButton.vue (hunk 3 above)
Comment (mahula) [reply presenter/src/components/inputs/MainButton.vue]: Good point.
Fixed in ba23543ae56bd8341e197230686bf2aa7f62ec11

File: presenter/src/components/inputs/MainButton.vue (hunk 4 above)
Comment (mahula) [reply presenter/src/components/inputs/MainButton.vue]: Done in ba23543ae56bd8341e197230686bf2aa7f62ec11

File: presenter/src/components/inputs/MainButton.vue (hunk 2 above)
Comment (mahula) [reply presenter/src/components/inputs/MainButton.vue:90]: A click is emitted. The 1 id an id value - see here https://github.com/dreammall-earth/dreammall.earth/blob/02fefe2f9603fa29947f3fbd902f0e019a57c8e8/presenter/src/components/inputs/MainButton.vue#L50

**Pull request merged**